crypto/keys/keys_test.go

```diff
@@ -138,21 +139,19 @@ func TestLoadPrivateKeyAndSign(t *testing.T) {
 		switch {
 		case test.keyPEM != "":
 			k, err = NewFromPrivatePEM(test.keyPEM, test.keyPass)
-			switch gotErr := err != nil; {
-			case gotErr != test.wantLoadErr:
-				t.Errorf("%s: NewFromPrivatePEM(_, _) = (%v, %v), want err? %t", test.name, k, err, test.wantLoadErr)
+			if got, want := errors.ErrorCode(err), test.wantLoadErr; got != want {
```

Comment:
Suggestion: Move the if outside the cases?

```go
var k crypto.Signer
var err error
switch {
case test.keyPEM != "":
  k, err = NewFromPrivatePEM(test.keyPEM, test.keyPass)
case test.keyPath != "":
  k, err = NewFromPrivatePEMFile(test.keyPath, test.keyPass)
}
if got, want := ... {
  t.Errorf("%v: NewFromPrivate() ...", ...)
}
```

There's a slight loss in the error message, but since we log the test name it should be fine. The readability / code duplication gain is significant, though.